src/main/java/com/klodnicki/Bike/v2/rest/controller/advice/GlobalCustomExceptionHandler.java

```diff
@@ -21,4 +20,9 @@ public ResponseEntity<String> handleMethodArgumentNotValidException(MethodArgume
         return new ResponseEntity<>("Your request miss required data, check your request with validation requirements",
                 HttpStatus.BAD_REQUEST);
     }
+
+    @ExceptionHandler(NotFoundInDatabaseException.class)
+    public ResponseEntity<?> handleNotFoundInDatabaseException(NotFoundInDatabaseException e) {
+        return new ResponseEntity<>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
yes, normally before I made this exception handler the status in that case was 500 Internal Server Error that's why I left how it was. 
Anyway I think that you are right, so i change it to 404 not found? why? because a resource is not found. In this case my resource is not a path but the object which was not found in database. the “resource” is the object I'm trying to retrieve from the database. If the object doesn’t exist, then from the client’s point of view, the “resource” is not found, even though the endpoint or path exists.